README.md

```diff
@@ -1,77 +1,151 @@
+
+RHEL8 (and newer) and Fedora no longer support RC4 encryption out of the box, it is recommended to enable AES in Active Directory, if not possible then the AD-SUPPORT crypto policy must be enabled.  The integration role will use the crypto_policies system role for this if the user sets the `ad_integration_manage_crypto_policies` and `ad_integration_allow_rc4_crypto` parameters to true.
+
+The Linux system must be able to resolve default AD DNS SRV records.
+
+Ports must be opened from Linux to AD:
+
+| Source Port | Destination | Protocol    | Service                                                   |
+|-------------|-------------|-------------|-----------------------------------------------------------|
+| 1024:65535  | 53          | TCP and UDP | DNS                                                       |
+| 1024:65535  | 389         | TCP and UDP | LDAP                                                      |
+| 1024:65535  | 636         | TCP         | LDAPS                                                     |
+| 1024:65535  | 88          | TCP and UDP | Kerberos                                                  |
+| 1024:65535  | 464         | TCP and UDP | Kerberos change/set password (kadmin)                     |
+| 1024:65535  | 3268        | TCP         | LDAP Global Catalog                                       |
+| 1024:65535  | 3269        | TCP         | LDAP Global Catalog SSL                                   |
+| 1024:65535  | 123         | UDP         | NTP/Chrony(Optional)                                      |
+| 1024:65535  | 323         | UDP         | NTP/Chrony (Optional)                                     |
+
 
 ## Role Variables
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+### Required variables
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+#### ad_integration_realm
 
-Example of setting the variables:
+Active Directory realm, or domain name to join
 
-```yaml
-template_foo: "oof"
-template_bar: "baz"
-```
+#### ad_integration_password
+
+The password of the user used to authenticate with when joining the machine to the realm.  Do not use cleartext - use Ansible Vault to encrypt the value.
+
+### Optional variables
+
+#### ad_integration_user
+
+The user name to be used to authenticate with when joining the machine to the realm.
+
+Default: Administrator
+
+#### ad_integration_join_to_dc
+
+an Active Directory domain controller's hostname (do not use IP address) may be specified to join via that domain controller directly.
+
+Default: Not set
```

Comment:
Is this a short name or an FQDN?  I know in the mssql role the short name had to be used in some cases - @spetrosi

Reply:
I don't think it matters as long as it is a DNS-resolveable name.

Reply:
Right, it should support both short and fqdn. mssql would need to use short names.

Reply:
Is this a hard requirement on the mssql side? I looked into this more and when we join with a specific AD DC **shortname** the join operation works fine, but SSSD fails to start the backend properly

~~~
[ad_sasl_log] (0x0040): SASL: GSSAPI Error: Unspecified GSS failure.  Minor code may provide more information (Server not found in Kerberos database)
~~~

SSSD relies on kerberos to communicate with AD, and kerberos relies on fully qualified names (at least without some additional configuration). Note this is not an issue if you join using the dns discovery based method using AD DNS SRV records, i.e. realm join AD.DOMAIN

Reply:
> Is this a hard requirement on the mssql side? I looked into this more and when we join with a specific AD DC **shortname** the join operation works fine, but SSSD fails to start the backend properly

Ok, then I guess it is not a hard requirement - we'll find out more when the mssql role is updated to use this role

> 
> ```
> [ad_sasl_log] (0x0040): SASL: GSSAPI Error: Unspecified GSS failure.  Minor code may provide more information (Server not found in Kerberos database)
> ```
> 
> SSSD relies on kerberos to communicate with AD, and kerberos relies on fully qualified names (at least without some additional configuration). This is why we can't use an IP address here, it must be a hostname. Note this is not an issue if you join using the dns discovery based method using AD DNS SRV records, i.e. realm join AD.DOMAIN